static/app/views/dashboards/detail.tsx

```diff
@@ -167,6 +170,23 @@ export function handleUpdateDashboardSplit({
   }
 }
 
+/* Checks if current user has permissions to edit dashboard */
+export function checkUserHasEditAccess(
+  dashboard: DashboardDetails,
+  currentUser: User,
+  organization: Organization
+): boolean {
+  if (organization.features.includes('dashboards-edit-access')) {
+    if (!dashboard.permissions) {
+      return true;
+    }
+    return dashboard.permissions.isCreatorOnlyEditable
+      ? dashboard.createdBy?.id === currentUser.id
+      : !dashboard.permissions.isCreatorOnlyEditable;
+  }
+  return true;
```

Comment:
This is a nit pick so it's optional feedback, but nesting adds a bit of complexity so it's preferable to exit early. It also allows us to combine the conditions together here

```suggestion
  if (!organization.features.includes('dashboards-edit-access') || !dashboard.permissions) {
    return true;
  }
  return dashboard.permissions.isCreatorOnlyEditable
    ? dashboard.createdBy?.id === currentUser.id
    : !dashboard.permissions.isCreatorOnlyEditable;
```